lib/configuration.ex

```diff
@@ -0,0 +1,89 @@
+defmodule Configuration do
+
+  def configure_game do
+    {player_1, player_2} = set_players()
+    board_size = get_board_size()
+    %Game{board: Board.create(board_size), 
+          size: board_size,
+          player_1: player_1,
+          player_2: player_2,
+          current_player: player_1}
+  end
+
+  def set_players(console \\ Console) do
+    game_type_input = console.get_game_type
+    if game_type_input |> Validator.is_valid_option? do
+      game_type_input 
+      |> Integer.parse 
+      |> elem(0) 
+      |> get_marks()
+    else
+      %Message{}.invalid |> console.print
+      set_players()
+    end
+  end
+  
+  def get_player_marks(player, console \\ Console) do
+    mark = player |> console.get_player_marks
+    if mark |> Validator.is_blank? do
+      %Message{}.cannot_be_blank
+      |> console.print
+      player
+      |> get_player_marks
+    else
+      if mark |> Validator.is_valid_length? do
+        mark
+      else
+        %Message{}.mark_length |> console.print
+        get_player_marks(player, console)
+      end
+    end
+  end
+
+  def get_marks(game_type, console \\ Console, console_2 \\ Console)
+
+  def get_marks(_human_vs_human = 1, console, console_2) do
+    player_1 = %Player{mark: get_player_marks(%Message{}.player_1, console)}
+    player_2 = %Player{mark: get_player_marks(%Message{}.player_2, console_2)}
+
+    if Validator.is_already_used?(player_1.mark, player_2.mark) do
+      %Message{}.cannot_match |> console.print
+      get_marks(1, console, console_2)
+    else
+      {player_1, player_2}
+    end
+  end
+
+
+  def get_marks(_humav_vs_computer = 2, console, _console) do
```

Comment:
I feel like your `mark` logic is good, but it might be a good opportunity for an extract module style refactor.  There's a lot of thought that you will need to organize for marks.